cpp/src/copying/copy.cpp

```diff
@@ -9,169 +9,186 @@
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either experimentalress or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
 
 #include <cudf/copying.hpp>
-#include <cudf/legacy/column.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/column/column.hpp>
 #include <utilities/column_utils.hpp>
 #include <utilities/error_utils.hpp>
 #include <cudf/cudf.h>
-#include <cudf/legacy/table.hpp>
+#include <cudf/table/table.hpp>
 #include <nvstrings/NVCategory.h>
 
 #include <cuda_runtime.h>
 #include <algorithm>
 
 namespace cudf
 {
+namespace experimental
+{
+namespace detail
+{
+
+inline mask_state should_allocate_mask(mask_allocation_policy mask_alloc, bool mask_exists) {
```

Comment:
just testing my understanding...

would it be appropriate to chance `bool mask_exists` to something like `mask_state existence` ?

Reply:
I will try to take care of it

Reply:
@cwharris `mask_state` is not associated with column and to get back the mask_state we have to again use `nullable()` and 'has_nulls()' to map it back to one of the mask_state. Which just increase steps and doesn't add much to code.